src/Helper/HeadMeta.php

```diff
@@ -458,10 +458,15 @@ public function set($value)
      * Not valid in a non-HTML5 doctype
      *
      * @param  string $charset
+     * @throws Exception\RuntimeException
      * @return HeadMeta Provides a fluent interface
      */
     public function setCharset($charset)
     {
+        if ($this->view->plugin('doctype')->isXhtml()) {
+            throw new Exception\RuntimeException('XHTML* doctype has no attribute charset');
```

Comment:
Please move this check to the `isValid` method and call `isValid` before `$this->set($item);` in the current method.

Reply:
You should also extend the exception message:

> 'XHTML* doctype has no attribute charset; please use appendHttpEquiv'

Then change the exception class to `Exception\InvalidArgumentException`.

Reply:
updated